parlai/tasks/self_chat/worlds.py

```diff
@@ -5,51 +5,111 @@
 # LICENSE file in the root directory of this source tree.
 
 import random
+from typing import Any, Dict, List
 
-from parlai.core.worlds import DialogPartnerWorld, validate
+from parlai.agents.repeat_label.repeat_label import RepeatLabelAgent
+from parlai.core.agents import Agent
+from parlai.core.worlds import create_task, DialogPartnerWorld, validate
 
 
-class InteractiveWorld(DialogPartnerWorld):
+def load_openers(opt):
+    base_task = opt['task'].split(':')[0]
+    if base_task == 'self_chat':
+        # TODO(2284): Load default openers from s3
+        return
+    print('[ loading conversation openers... ]')
+    # create dummy task so we can get openers from the data
+    task_opt = opt.copy()
+    task_opt['task'] = base_task
+    # default train will loop forever, but evalmode will stop after one epoch
+    if task_opt['datatype'].startswith('train'):
+        task_opt['datatype'] = 'train:evalmode'
+    task_opt['interactive_task'] = False
+    task_agent = RepeatLabelAgent(task_opt)
+    task_world = create_task(task_opt, task_agent)
+    # run through task data, collecting all first messages
+    openers = set()
+    is_first_turn = True
+    while not task_world.epoch_done():
+        task_world.parley()
+        msg = task_world.get_acts()[0]
+        # add only the first message in the episode
+        if is_first_turn:
+            openers.add(msg.get('text', ''))
```

Comment:
I'd worry about getting the empty string in here. If there's no text, I'd rather add skip it.